### ISSUE TITLE
Stop Motors When Not in Use While Running

### ISSUE THREAD

**Issue by krkeegan** [MaslowCNC/Firmware#270]
This issue is most apparent currently when using the zaxis.  While plunging in or retracting, you can audibly hear the left and right motors twitching trying to move to their exact assigned position, overshooting and moving back.  Generally these are extremely subtle movements, but not always.

However, similarly, but less noticeable, is that the zaxis motor continues to twitch as well trying to hold its assigned position when making a coordinated XY movement.

In an ideal world where we have worked out all of the bugs, the risks from this activity are low.  The machine should be able to tweak all three motors constantly.  Still, this may lead to unnecessary heat on the motor controllers and I could imagine it causing other electrical issues.  Plus our world is not ideal, and twitching these motors unnecessarily can lead to other issues.

Within CNCFunctions we call `axis.attach()` before moving an axis.  In short, this "turns on" the motor, when attached, the PID process will constantly fight to get the motor to its exact assigned position.  If we call `axis.detach()` the PID process no longer tries to control the motor and we cut power to the motor.

We rarely call `axis.detach()`, I don't think it ever gets called when processing a normal stream of gcode commands.  So while all axes start out detached, once they are used for the first time, they remain active until the gcode stream is stopped or completed.

Sadly, we can't simply call `axis.detach()` at the end of a function such as `cordinatedMove` or `arc` because the motor may not have completed its final movement when these functions complete.

So how do we turn off select motors? Some of my ideas, I am looking for an opinion on what the best course of action is:

1. Detach at the end of `cordinatedMove` and similar functions, but only after the `axis.error` has entered some tolerance.
2. Leave `cordinatedMove`, `arc` and any other function which might some day allow for simultaneous three axis movement alone.  Only alter `singleAxisMove` and any other function that specifically envisions that some axes would be locked.  These functions would start with a check of the other two axes and loop until after these `axis.error` has entered some tolerance.  Then the single axis would perform its action.
3. Move the solution down into the `axis` class.  Such as after `axis.endMove` has been called another function would be triggered to monitor `axis.error` and would detach the axis after it has entered some tolerance.  Perhaps also waiting some temporal delay as well.

I am a little unsure what would happen in 1 if there was a series of `cordinatedMove` calls in a row, I don't know if things would become less smooth because the motor is being turned off and on frequently.

1 & 2 leave open the possibility that we could get stuck if the tolerance is set too high, but this could likely be solved with a maximum time. 

3 seems a bit odd, because a higher level function would need to recognize that an axis could be detached out from underneath it.

Sorry for the wall of words.

**Comment by krkeegan** [comment]
I vote to close this.

Previously the bad PID values caused the motors to jump.  Once those were changes I see that the motors are more than capable of holding station when powered.
